spec/unit/lab_manager/provider/vsphere_spec.rb

```diff
@@ -0,0 +1,202 @@
+      dest_folder: 'zoo/xoo/coo',
+      cluster: 'kokoko',
+      power_on: true
+    }
+  end
+
+  let(:vm_clone_response) do
+    {
+      'power_state' => 'poweredOn',
+      'vm_ref' => 'xxxx',
+      'new_vm' => { 'id' => SecureRandom.uuid }
+    }
+  end
+
+  let!(:provider) do
+    Provider::VSphere.new(build(:compute, provider_name: :v_sphere, name: 'foo'))
+  end
+
+  before(:each) do
+    allow(Provider::VSphereConfig).to receive(:create_vm_defaults) { {} }
+    allow(Provider::VSphere).to receive(:connect) { connection_pool_mock }
+    allow(connection_pool_mock).to receive(:with).and_yield(vsphere_mock)
+    allow(Provider::VSphereConfig).to receive(:create_vm_defaults) { vm_defaults }
+  end
+
+  describe '#create_vm' do
+    it 'arguments are propagated to fog#clone_vm' do
+      args = {
+        name: 'a',
+        datacenter: 'b',
+        template_path: 'c',
+        cluster: 'd',
+        linked_clone: true,
+        power_on: true,
+        dest_folder: 'z'
+      }
+      allow(vsphere_mock).to receive(:vm_clone) do |param|
+        expect(param['name']).to eq args[:name]
+        expect(param['datacenter']).to eq args[:datacenter]
+        expect(param['template_path']).to eq args[:template_path]
+        expect(param['cluster']).to eq args[:cluster]
+        expect(param['linked_clone']).to eq args[:linked_clone]
+        expect(param['power_on']).to eq args[:power_on]
+        expect(param['dest_folder']).to eq args[:dest_folder]
+
+        vm_clone_response
+      end
+
+      allow(provider).to receive(:power_on) {}
+      provider.create_vm(args)
+    end
+
+    it 'default arguments are propagated to fog#clone_vm as well' do
+      allow(vsphere_mock).to receive(:vm_clone) do |param|
+        expect(param['datacenter']).to eq 'doo'
+        expect(param['template_path']).to eq 'aoo/boo/coo'
+        expect(param['cluster']).to eq 'kokoko'
+        expect(param['linked_clone']).to eq true
+        expect(param['power_on']).to eq true
+        expect(param['dest_folder']).to eq 'zoo/xoo/coo'
+
+        vm_clone_response
+      end
+
+      allow(provider).to receive(:power_on) {}
+      provider.create_vm
+    end
+
+    it 'name is generated automatically when not provided ' do
+      allow(vsphere_mock).to receive(:vm_clone) do |param|
+        expect(param['name']).to match(/^lm/)
+
+        vm_clone_response
+      end
+
+      allow(provider).to receive(:power_on) {}
+      provider.create_vm
+    end
+
+    it 'add_machine_to_drs_rule is called when requested' do
+      allow(vsphere_mock).to receive(:vm_clone) { vm_clone_response }
+
+      allow(provider).to receive(:power_on) {}
+      allow(provider).to receive(:add_machine_to_drs_rule_impl) do |_obj, params|
+        expect(params[:group]).to eq 'foooo'
+      end
+
+      allow(provider).to receive(:machine_present_in_drs_rule?) do |_obj, params|
+        expect(params[:group]).to eq 'foooo'
+      end
+
+      provider.create_vm
```

Comment:
I guess this test will be fail, you have to add `allow(Provider::VSphereConfig.create_vm_defaults).to receive(:add_to_drs_group) { ... }`